src/Modules/CalcSetup.lua

```diff
@@ -759,6 +765,24 @@ function calcs.initEnv(build, mode, override, specEnv)
 		end
 	end
 
+	-- Add granted ascendancy node (e.g., Forbidden Flame/Flesh combo)
+	local matchedName = { }
+	for _, ascTbl in pairs(env.modDB:List(nil, "GrantedAscendancyNode")) do
+		local name = ascTbl.name
+		if matchedName[name] and matchedName[name].side ~= ascTbl.side and matchedName[name].matched == false then
+			matchedName[name].matched = true
+			local node = env.spec.tree.ascendancyMap[name]
+			if node and (not override.removeNodes or not override.removeNodes[node.id]) then
+				if env.itemModDB.conditions["ForbiddenFlesh"] == env.spec.curClassName and env.itemModDB.conditions["ForbiddenFlame"] == env.spec.curClassName then
+					env.allocNodes[node.id] = node
+					env.grantedPassives[node.id] = true
+				end
+			end
```

Comment:
If you take my suggestion from above for never adding the mods in the first place, this code is all duplicated with "GrantedPassive".  I've been trying to think of a way to cleanly integrate the two, but I'm out of time to actually test it myself.  I think you can do something like this in ModParser: `["allocates (.+) if you have the matching modifiers on (.+)"] = function(_, passive, matching) return { mod("GrantedPassive", "LIST", passive, { type = "Condition", var = "has" .. matching }) } end,`  I wish there was a way to see the source of a mod here, as that would make this a lot cleaner.

This code block is fine if there isn't another way, but having "GrantedAscendancyNode" when that could work the same as "GrantedPassive" + a condition feels wrong

Reply:
Keeping it as is for now as I'm out of time myself.